yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,822 @@
+    container domain-password {
+      description "domain password.";
+      configd:help "domain password.";
+      uses isis-password ;
+    }
+  }
+
+  grouping redistribute-attributes {
+    configd:help "Common optional attributes of any redistribute entry.";
+    description  "Common optional attributes of any redistribute entry.";
+    choice attribute {
+      list route-map {
+        configd:help "Route map reference.";
+        description "Route map reference.";
+        key "rm-name";
+        leaf rm-name {
+          type string {
+            length "1..max";
+          }
+          configd:help "Applies the conditions of the specified route-map to routes that
+          are redistributed into this routing instance.";
+          description "Applies the conditions of the specified route-map to routes that
+          are redistributed into this routing instance.";
+        }
+        leaf metric {
+          configd:help "Metric for default route.";
+          description "Metric for default route.";
+          type uint32 {
+            range "0..16777215";
+          }
+          default "0";
+        }
+      }
+      list metric {
+        configd:help "Metric for default route.";
+        description "Metric for default route.";
+        key "metric-id";
+        leaf metric-id {
+          type uint32 {
+            range "0..16777215";
+          }
+          default "0";
+          configd:help "Metric used for the redistributed route. If 0,
+          the default-metric attribute is used instead.";
+          description "Metric used for the redistributed route. If 0,
+          the default-metric attribute is used instead.";
+        }
+        leaf route-map {
+          type string {
+            length "1..max";
+          }
+          configd:help "Route map reference.";
+          description "Route map reference";
+        }
+      }			
+    }
+  }
+
+  grouping redistribute-default {
+    container always {
+      configd:help "Always advertise default route.";
+      description "Always advertise default route.";
+      uses redistribute-attributes;
+    }
+    uses redistribute-attributes;
+  }
+
+
+  grouping isis-net {
+    container default-information {
+      configd:help "Distribution of default information.";
+      description "Distribution of default information.";
+      container originate {
+        configd:help "Distribute a default route.";
+        description "Distribute a default route.";
+        list ipv4 {
+          key "level";
+          configd:help "Distribute default route for IPv4.";
+          description "Distribute default route for IPv4.";
+          leaf level {
+            type level;
+          }
+        }
+
+        list ipv6 {
+          key "level";
+          configd:help "Distribute default route for IPv6.";
+          description "Distribute default route for IPv6.";
+          leaf level {
+            type level;
+          }
+        }
+      }
+    }
+    leaf net {
```

Comment:
The description suggests this is a list of addresses, but a leaf will only allow a single value. Should this be a leaf-list instead?

Reply:
This is Resolved.